cpp/src/gandiva/decimal_type_util.h

```diff
@@ -24,12 +24,14 @@
 #include <memory>
 
 #include "gandiva/arrow.h"
+#include "gandiva/visibility.h"
 
 namespace gandiva {
+namespace internal {
```

Comment:
why make this internal ? I was hoping that callers can use this to determine the output type given the input types.

Reply:
I was dealing with a visibility issue that I've since resolved (and DecimalTypeUtil is exported again). I'll remove the `internal` namespace